client/src/services/api.js

```diff
@@ -0,0 +1,45 @@
+const baseUrl = "/api/v1"
+
+const fetchData = async (url, options) => {
+    try {
+        const response = await fetch(url, options)
+        if (!response.ok) {
+            const errorBody = await response.json()
+            if (response.status === 422) {
+                return errorBody
+            } else {
+                throw new Error(`Error in fetch: ${response.status} (${response.statusText}) ${errorBody.errors}`)
+            }
+        }
+        return (await response.json())
+    } catch (err) {
+        return console.error(err.message)
+    }
+}
+
+export const fetchDialogs = async (userId) => {
+    const url = userId ? `${baseUrl}/dialogs?user_id=${userId}` : `${baseUrl}/dialogs`
```

Comment:
this is either over engineered, or needs to be refactored based on actual implementation. The above use of `fetchDialogs` does not pass a `userId` as an argument, and therefore should be removed. If the intended idea is for the current user to get _their_ associated dialogs, then we should let the back end server access the currently signed in user `req.user` to grab their dialogs